### DISCUSSION TITLE
Why do I connect two OFFRFobjects with GenericJoint and they still separate in the simulation?

### DISCUSSION THREAD

**Discussion by qiushi-Lee** [jgerstmayr/EXUDYN#27] [Q&A on usage of Exudyn] [answered]
As stated in the title, I have connect two OFFRFobjects by GenericJoint and constraint all degree of freedom as constrainAxes = [1,1,1,1,1,1] using MarksuperelementRigid. However, when I added load and started dynamicsolver, the two OFFRFobjects were still separated during the simulation. I have set the offset of the marker, but the Axes of two object were different as the following photo. Does that influent the simulation result? How to deal with this problem?
Thanks a lot for your kind response.

![image](https://user-images.githubusercontent.com/111936972/192490633-db04a664-5f37-4646-8bab-b94b7cbf93c5.png)

**Comment by jgerstmayr** [comment, marked as answer]
If you use a GenericJoint, it connects two rigid body markers. They need to have same orientation and position initially. If you want to connect them with different local frames, then you need to add a markerRotation at the GenericJoint (and possibly and offset in the joint).
I assume that you use the index2 solver: in this case, different initial orientation will lead to arbitrary solution. If you use generalized alpha, it would try to immediately rotate and move the bodies such that the two frames of the GenericJoint match (which may cause severe problems for the Newton solver).

**Reply by qiushi-Lee** [reply]
I‘m sorry to bother you again. I have set the GenericJoint as your statement. However, the two objects still separate at the GenericJoint. I have attempted to add RigidBodyMotion to OFFRFobject, but it didn't work. I  didn't know how to solve that problem.
![image](https://user-images.githubusercontent.com/111936972/216207692-50365a79-93f1-454d-8677-3d4b3da17c56.png)
![image](https://user-images.githubusercontent.com/111936972/216207748-427df78d-6e5b-4d41-af0e-38844ff64e6c.png)

**Reply by jgerstmayr** [reply]
I have no particular idea where the exact problem lies. There are examples of ObjectFFRF (and in particular ObjectFFRFreducedOrder which I recommend). Can you replicate those examples? In case of ObjectFFRF you need to fix the reference frame as done in the examples. 
In any case, a minimum example code would help to analyze your problem. You could just send it to me zipped via email.
